src/components/calcite-input/calcite-input.tsx

```diff
@@ -183,27 +240,270 @@ export class CalciteInput {
   }
 
   componentWillLoad(): void {
-    this.defaultValue = this.value;
     this.childElType = this.type === "textarea" ? "textarea" : "input";
-    this.requestedIcon = setRequestedIcon(INPUT_TYPE_ICONS, this.icon, this.type);
-    this.minString = this.min?.toString();
+    this.defaultValue = this.value;
     this.maxString = this.max?.toString();
-    this.stepString = this.step?.toString();
+    this.minString = this.min?.toString();
+    this.requestedIcon = setRequestedIcon(INPUT_TYPE_ICONS, this.icon, this.type);
   }
 
   componentDidLoad(): void {
     this.slottedActionEl = this.el.querySelector("[slot=input-action]");
     this.setDisabledAction();
+    if (this.shouldFormatNumberByLocale()) {
+      this.childEl.style.cssText = hiddenInputStyle;
```

Comment:
Does this need to happen in another lifecycle hook? `componentDidLoad` only happens once, so this could be problematic if the component is reused.

Reply:
Unfortunately, `this.childEl` isn't accessible until the first `render` cycle, so this operation can't be performed inside `connectedCallback` or `componentWillLoad` or `componentWillRender` (on the first render cycle) so the only place that made sense to me to do this was in `componentDidLoad`.

Reply:
Just an idea (for later), but If `hiddenInputStyle` was an object, we could set it directly on the `style` prop where childEl is rendered. This would eliminate the need to get a reference to the node and set it.